lib/active_model_serializers/key_transform.rb

```diff
@@ -66,5 +66,26 @@ def underscore(value)
     def unaltered(value)
       value
     end
+
+    # Transforms string to selected case
+    # Accepts string in any case: 'camel', 'undescore', 'dashed'.
+    #
+    # @example:
+    #   transform('SomeClass', :underscore) => 'some_class'
+    #   transform('some_class', :snake) => 'someClass'
+    #   etc...
+    def transform(string, type)
+      string = string.underscore
+      case type.to_sym
+      when :dashed
+        string.dasherize
+      when :camel
+        string.camelize
+      when :snake
+        string.camelize(:lower)
+      else
+        string
+      end
+    end
```

Comment:
- Is there a reason this doesn't leverage the existing transforms?
- Why does there need to be a `jsonapi_type_transform` and a `key_transform`? Are there situations you'd really want them to differ?
- Can `type.map! ...` from `lib/active_model_serializers/adapter/json_api/resource_identifier.rb` just use `KeyTransform.send`?
- Isn't "snake (underscore)" `snake_case`, "camel (lower)" `camelCase`, and camel (pascal) `CamelCase`?